cryostatio/cryostat-web
# feat(dashboard): notify user when cards cannot be loaded

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Fixes: #876 
Related to #950 

## Description of the change:

- [x] Added a validator for card configurations when build a card. 
- [x] Fixed dark mode text color in dashboard solo.

**Note**: prop value validation ignores dynamic list (i.e. via Observable). Though there should be a way to fetch options and check but with current setup, refactoring won't come in time for code freeze.

## Motivation for the change:

See #876 

## How to manually test:
1. Run `CRYOSTAT_IMAGE=quay.io... sh smoketest.sh...`
2. Mess around `DASHBOARD_CFG` entry in local storage (e.g. try using an invalid prop value, card name).
3. Reload to see the error card (previously the whole UI will crash).

## Screenshots

| Light Theme | Dark Theme |
|--|--|
|![Screenshot from 2023-04-13 21-16-18](https://user-images.githubusercontent.com/68053619/231916580-6f5378c4-e965-4c23-916c-174ce979e44d.png)|![Screenshot from 2023-04-13 21-27-22](https://user-images.githubusercontent.com/68053619/231917647-158ad03a-82fe-4d61-9094-46fa9f007254.png)|
